app/screens/login/login.js

```diff
@@ -0,0 +1,173 @@
+    });
+  }
+
+  componentDidMount() {  
+    this._presetLogin();
+    this._setupGoogleSignin();
+  }
+
+  _login = () => {
+    Keyboard.dismiss();
+    this._setErrorMessage();    
+    if (this.state.email === loginCredentials.email && this.state.password === loginCredentials.password) {
+      startTabScreen();      
+    } else {
+      // incorrect email / password
+      this._setErrorMessage('Incorrect Credentials!');      
+    }
+  }
+
+  async _setupGoogleSignin() {
+    try {
+      await GoogleSignin.hasPlayServices({ autoResolve: true });
+      await GoogleSignin.configure({
+        iosClientId: IOS_GOOGLE_CLIENT_ID,
+        offlineAccess: false
+      });
+
+      const user = await GoogleSignin.currentUserAsync();
+      console.log(user);
+      this.setState({user});
+      this.props.onLogin(user);      
+    }
+    catch(err) {
+      console.log("Play services error", err.code, err.message);
+    }
+  }
+
+  _googleSignIn() {
+    GoogleSignin.signIn()
+    .then((user) => {
+      this.setState({user: user});
+      this.props.onLogin(user);
+    })
+    .catch((err) => {
+      console.log('WRONG SIGNIN', err);
+    })
+    .done();
+  }
+
+  _validateEmail = (email) => {
+    isEmailValid = validateEmail(email);
+    if (!isEmailValid) {
+      this._setErrorMessage('Invalid Email!');
+    }
+  }
+
+  _setErrorMessage = (message) => {
+    this.setState({ errorMessage: message })
+  }
+
+  render() {
+    return (
+      <View style={style.mainContainer}>
+        <View style={style.logoContainer}>
+          <Image source={logo} style={style.logoImage}/>
+        </View>
+        <View style={style.formContainer}>
+          <View style={style.errorLabelContainer}>
+            <Text style={style.errorLabel}>{ this.state.errorMessage }</Text>
+          </View>
+          <View style={style.formSubContainer}>
+            <View style={style.emailFieldWrapper}>
+              <TextInput
+                ref={(component) => this.emailField = component }
+                placeholder='Email'
+                returnKeyType='next'
+                autoCapitalize='none'
+                autoCorrect={ false }
+                style={ style.emailField }
+                keyboardType='email-address'
+                onFocus={() => this._setErrorMessage()}
+                onChangeText={
+                  (text) => {
+                    this.setState({ email: text });
+                  }
+                }
+                underlineColorAndroid='#bbb'
+                placeholderTextColor='#bbb'
+                onSubmitEditing={ () => {
+                  this._validateEmail(this.state.email);                  
+                  this.passwordField.focus()
+                }}
+                value={this.state.email}
+              />
+            </View>
+            <View style={style.passwordFieldWrapper}>
+              <TextInput
+                ref={(component) => this.passwordField = component }
```

Comment:
follow js coding guidelines from the link mentioned above

Reply:
🆗